server/aws-lsp-codewhisperer/src/language-server/agenticChat/agenticChatController.ts

```diff
@@ -1104,12 +1144,23 @@ export class AgenticChatController implements ChatHandlers {
             this.#telemetryController.emitMessageResponseError(tabId, metric.metric, err.requestId, err.message)
         }
 
-        if (err instanceof AmazonQServicePendingSigninError) {
+        // return non-model errors back to the client as errors
+        if (!(err instanceof ModelServiceException)) {
```

Comment:
would an auth error end up going through this flow? Since it's not a model service exception

Reply:
auth errors come from the SDK request, so they are thrown by `fetchModelRequest` and thus wrapped as well.